dbms/src/DataStreams/TiRemoteBlockInputStream.h

```diff
@@ -195,6 +195,15 @@ class TiRemoteBlockInputStream : public IProfilingBlockInputStream
         if (kill)
             remote_reader->cancel();
     }
+
+    void readPrefixImpl() override
+    {
+        if constexpr (std::is_same_v<RemoteReader, CoprocessorReader>)
+        {
+            remote_reader->open();
+        }
```

Comment:
How about moving to `readImpl` like https://github.com/pingcap/tiflash/blob/7203151a59ac7c44c673fc5f689fef312c660731/dbms/src/Storages/DeltaMerge/ReadThread/UnorderedInputStream.h#L82?
It will be executed later.

Reply:
I think PreadrefixImpl is more appropriate, it may send requests earlier in some scenarios, but it can reduce the latency of reading。

Reply:
Add a check in `readImpl` to make sure the reader is already opened?

Reply:
considered, seems resp.iter's next will report when it wasn't open, let me tested it.

Reply:
> Add a check in `readImpl` to make sure the reader is already opened?

added